app/views/leaderboard.scala.html

```diff
@@ -6,7 +6,13 @@
 @leaderboardStatsThisWeek = @{UserStatTable.getLeaderboardStats(10, "weekly")}
 @leaderboardStatsByOrg = @{UserStatTable.getLeaderboardStats(10, "overall", true)}
 @leaderboardStatsOrg = @{UserStatTable.getLeaderboardStats(10, "overall", false, UserOrgTable.getAllOrgs(user.get.userId).headOption)}
+@import play.api.Play
+@import play.api.Play.current
+@import models.utils.Configs
 
+@currentCityId =@{Play.configuration.getString("city-id").get}
+@cityInfo = @{Configs.getAllCityInfo(lang).filter(c => c.cityId == currentCityId || c.visibility == "public" || (List("taipei", "new-taipei-tw", "keelung-tw").contains(currentCityId) && List("taipei", "new-taipei-tw", "keelung-tw").contains(c.cityId)))}
```

Comment:
All you need is the country id for the current city, so you should just get that instead of using this to get the info for every city!

Reply:
Got it, went ahead and fixed that